doc/INTERNALS.md

```diff
@@ -0,0 +1,99 @@
+# Internals
```

Comment:
This looks like it comments the code. Why is it not in the uid header? It's a bit hidden here, someone reading the uid header may not look for documentation here.